src/main.cc

```diff
@@ -407,327 +928,183 @@ static void bro_new_handler()
-		return context.run();
-#endif
-		}
-	else
-		{
-		bro_argc = argc;
-		bro_argv = new char* [argc];
-
-		for ( int i = 0; i < argc; i++ )
-			bro_argv[i] = copy_string(argv[i]);
-		}
-
-	double time_start = current_time(true);
-
-	brofiler.ReadStats();
+	std::vector<std::string> rval;
 
-	name_list interfaces;
-	name_list read_files;
-	name_list rule_files;
-	char* id_name = 0;
-
-	char* seed_load_file = zeekenv("ZEEK_SEED_FILE");
-	char* seed_save_file = 0;
-	char* user_pcap_filter = 0;
-	char* debug_streams = 0;
-	int parse_only = false;
-	int bare_mode = false;
-	int do_watchdog = 0;
-	int override_ignore_checksums = 0;
-	int rule_debug = 0;
-	int RE_level = 4;
-	int print_plugins = 0;
-	int time_bro = 0;
-
-	static struct option long_opts[] = {
-		{"parse-only",	no_argument,		0,	'a'},
-		{"bare-mode",	no_argument,		0,	'b'},
-		{"debug-policy",	no_argument,		0,	'd'},
-		{"exec",		required_argument,	0,	'e'},
-		{"filter",		required_argument,	0,	'f'},
-		{"help",		no_argument,		0,	'h'},
-		{"iface",		required_argument,	0,	'i'},
-		{"zeekygen",		required_argument,		0,	'X'},
-		{"prefix",		required_argument,	0,	'p'},
-		{"readfile",		required_argument,	0,	'r'},
-		{"rulefile",		required_argument,	0,	's'},
-		{"tracefile",		required_argument,	0,	't'},
-		{"writefile",		required_argument,	0,	'w'},
-		{"version",		no_argument,		0,	'v'},
-		{"no-checksums",	no_argument,		0,	'C'},
-		{"force-dns",		no_argument,		0,	'F'},
-		{"load-seeds",		required_argument,	0,	'G'},
-		{"save-seeds",		required_argument,	0,	'H'},
-		{"print-plugins",	no_argument,		0,	'N'},
-		{"prime-dns",		no_argument,		0,	'P'},
-		{"time",		no_argument,		0,	'Q'},
-		{"debug-rules",		no_argument,		0,	'S'},
-		{"re-level",		required_argument,	0,	'T'},
-		{"watchdog",		no_argument,		0,	'W'},
-		{"print-id",		required_argument,	0,	'I'},
-		{"status-file",		required_argument,	0,	'U'},
-
-#ifdef	DEBUG
-		{"debug",		required_argument,	0,	'B'},
-#endif
-#ifdef	USE_IDMEF
-		{"idmef-dtd",		required_argument,	0,	'n'},
-#endif
-#ifdef	USE_PERFTOOLS_DEBUG
-		{"mem-leaks",	no_argument,		0,	'm'},
-		{"mem-profile",	no_argument,		0,	'M'},
-#endif
-
-		{"pseudo-realtime",	optional_argument, 0,	'E'},
-		{"test",		no_argument,		0,	'#'},
-
-		{0,			0,			0,	0},
-	};
-
-	enum DNS_MgrMode dns_type = DNS_DEFAULT;
-
-	dns_type = bro_dns_fake() ? DNS_FAKE : DNS_DEFAULT;
-
-	RETSIGTYPE (*oldhandler)(int);
-
-	prog = argv[0];
-
-	prefixes.push_back(strdup(""));	// "" = "no prefix"
+	// Parse rule files defined on the script level.
+	char* script_signature_files =
+		copy_string(internal_val("signature_files")->AsString()->CheckString());
 
-	char* p = zeekenv("ZEEK_PREFIXES");
+	char* tmp = script_signature_files;
+	char* s;
+	while ( (s = strsep(&tmp, " \t")) )
+		if ( *s )
+			rval.emplace_back(s);
 
-	if ( p )
-		add_to_name_list(p, ':', prefixes);
+	delete [] script_signature_files;
+	return rval;
+	}
 
-	string zeekygen_config;
+static std::string get_exe_path(const std::string& invocation)
```

Comment:
Could this just use `realpath`?

Reply:
Don't think so -- the more important thing being done here is the `PATH` lookup.  e.g. find out to what an invocation via just `zeek` refers.  Nice to preserve the absolute path in case it needs to `exec()` a new Stem since it's otherwise just another way to pick up a different binary (if a different one got inserted at an early point in `PATH` before the `exec()` happens)